conf/application.conf

```diff
@@ -35,20 +46,23 @@ play.modules {
 akka.grpc.client {
 
   "helloworld.GreeterService" {
-    # configure ssl to trust our self signed certificate
-    host = localhost
-    port = 9443
+    # default `host` to the address where the server's HTTPS endpoint was bound but use the `DEPLOYMENT_SERVICE_NAME`
+    # when available.
+    host = ${play.server.https.address}
+    host = ${?DEPLOYMENT_SERVICE_NAME}
+    port = ${play.server.https.port}
+    use-tls = true
+    # The Authority on the requests must match the CN on the server certificate. The certificate on
+    # `conf/selfsigned.keystore` was issued to `localhost` so we must override the authority.
+    override-authority = "localhost"
```

Comment:
```
    host = ${play.server.https.address}
    host = ${?DEPLOYMENT_SERVICE_NAME}
    port = ${play.server.https.port}
```
This took a bit of trial-and-error... The key is to reuse `play.server.https.*` settings but prefer `DEPLOYMENT_SERVICE_NAME` if available.

The `override-authority` is required because of the `CN` on the fake cert.